MdsSupportingClasses/Collivery.php

```diff
@@ -945,141 +966,53 @@ public function addContact(array $data)
 	 */
     public function getPrice(array $data)
     {
-        $towns = $this->getTowns();
-
-        if (!isset($data['collivery_from']) && !isset($data['from_town_id'])) {
-            $this->setError('missing_data', 'collivery_from/from_town_id not set.');
-        } elseif (isset($data['collivery_from']) && !is_array($this->getAddress($data['collivery_from']))) {
-            $this->setError('invalid_data', 'Invalid Address ID for: collivery_from.');
-        } elseif (isset($data['from_town_id']) && !isset($towns[$data['from_town_id']])) {
-            $this->setError('invalid_data', 'Invalid Town ID for: from_town_id.');
+        // HERE WE GO
+        $towns = $this->make_key_value_array($this->getTowns(), 'id', 'name');
+
+        if (!isset($data['collection_town']) && !isset($data['collection_address'])) {
+            $this->setError('missing_data', 'collection_town/collection_address not set.');
+        } elseif (isset($data['collection_address']) && !is_array($this->getAddress($data['collection_address']))) {
+            $this->setError('invalid_data', 'Invalid Address ID for: collection_address.');
+        } elseif (isset($data['collection_town']) && !isset($towns[$data['collection_town']])) {
+            $this->setError('invalid_data', 'Invalid Town ID for: collection_town.');
         }
 
-        if (!isset($data['collivery_to']) && !isset($data['to_town_id'])) {
-            $this->setError('missing_data', 'collivery_to/to_town_id not set.');
-        } elseif (isset($data['collivery_to']) && !is_array($this->getAddress($data['collivery_to']))) {
-            $this->setError('invalid_data', 'Invalid Address ID for: collivery_to.');
-        } elseif (isset($data['to_town_id']) && !isset($towns[$data['to_town_id']])) {
-            $this->setError('invalid_data', 'Invalid Town ID for: to_town_id.');
+        if (!isset($data['delivery_address']) && !isset($data['delivery_town'])) {
+            $this->setError('missing_data', 'delivery_address/delivery_town not set.');
+        } elseif (isset($data['delivery_address']) && !is_array($this->getAddress($data['delivery_address']))) {
+            $this->setError('invalid_data', 'Invalid Address ID for: delivery_address.');
+        } elseif (isset($data['delivery_town']) && !isset($towns[$data['delivery_town']])) {
+            $this->setError('invalid_data', 'Invalid Town ID for: delivery_town.');
         }
 
-        if (!isset($data['service'])) {
+        if (!isset($data['services'])) {
             $this->setError('missing_data', 'service not set.');
         }
 
         if (!$this->hasErrors()) {
```

Comment:
Instead of all of the following code being wrapped in an `if()` perhaps you could could reverse the check and return early:
```php
if ($this->hasErrors()) {
  return;
}
```

Reply:
Updated accordingly.